superset-frontend/src/SqlLab/types.ts

```diff
@@ -53,6 +56,21 @@ export type toastState = {
   noDuplicate: boolean;
 };
 
+export type UnsavedQueryEditor = Partial<QueryEditor> & {
+  id?: QueryEditor['id'];
+};
+
+export interface Table {
+  id: string;
+  dbId: number;
+  schema: string;
+  name: string;
+  queryEditorId: QueryEditor['id'];
+  dataPreviewQueryId?: string | null;
```

Comment:
Do you explicitly need `null` or can we use `undefined`?
```suggestion
  dataPreviewQueryId?: string;
```